### PR TITLE
Tradingpost

### PR DESCRIPTION
**Pull request by n1tr0-5urf3r**
Added trading post commands for transactions 
invoke with !tp current buys/sells

Returns up to 20 current listings with quantity, your price and maximum price offer at the moment

### CODE REVIEW

**Comment by Maselkov** [comment]
You're doing good work. Will review it tomorrow.

**Comment by n1tr0-5urf3r** [comment]
Thanks! It is my first python project ever

File: guildwars2/guildwars2.py [hunk 1]
```diff
@@ -1207,6 +1207,69 @@ def __unload(self):
...
+            await send_cmd_help(ctx)
+
+    @tp.command(pass_context=True)
+    async def current(self, ctx, state):
+        """Show current selling/buying transactions
+        invoke with sells or buys"""
+
+        user = ctx.message.author
+        color = self.getColor(user)
+        state = state.lower
```
Comment (Maselkov) [inline guildwars2/guildwars2.py]: Needs brackets after lower
```
state = state.lower()
```

**Review by Maselkov — CHANGES_REQUESTED** [review]
It's good but It's making unnecessary API calls - 
* The items endpoint can be requested with ids= parameter, to get multiple items at once (each id separated by comma, see https://wiki.guildwars2.com/wiki/API:2/items)
* Same thing with commerce/listings, you can use ids=

File: guildwars2/guildwars2.py (hunk 1 above)
Comment (n1tr0-5urf3r) [reply guildwars2/guildwars2.py]: I'll look into that

**Comment by n1tr0-5urf3r** [comment]
Seems to work, API is only called once now for listings and items

**Pull request merged**